pip/pip-375.md

```diff
@@ -0,0 +1,41 @@
+# PIP-375: Expose the Admin client configs: readTimeout, requestTimeout, connectionTimeout
+
+# Background knowledge
+
+- `pulsar-admin sinks create` will upload files to brokers, but if the file is large and the network is slow, it may cost a long time.
+- The default value of `requestTimeout` of Admin API is `60s`, after the fix https://github.com/apache/pulsar/pull/23128, it was changed to `300s`.
+
+# Motivation
+
+We'd better to expose the following configs to users:
+- `readTimeout`
+- `requestTimeout`
+- `connectionTimeout`
+
+# Goals
+
+Expose the Admin client configs: `readTimeout`, `requestTimeout` and `connectionTimeout` to users.
+
+# Detailed Design
+
+### Configuration
+
+**broker.conf**
+```properties
+brokerAdminReadTimeoutInSeconds=120
+brokerAdminRequestTimeoutInSeconds=120
+brokerAdminConnectionTimeoutInSeconds=30
+```
+
+### CLI
+
+**client.conf**
+```properties
+adminReadTimeoutInSeconds=300
+adminRequestTimeoutInSeconds=300
+adminConnectionTimeoutInSeconds=30
+```
```

Comment:
Are these values intended to be good sample values? 
In other sources, I think that read timeout is typically set to a rather short timeout.

Reply:
Restored to the value the same as the current default configuration.